sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosResponseDiagnosticsTest.java

```diff
@@ -59,23 +59,23 @@ public void afterClass() throws CosmosClientException {
     @Test(groups = {"simple"})
     public void gatewayDiagnostics() throws CosmosClientException {
         CosmosItemProperties cosmosItemProperties = getCosmosItemProperties();
-        CosmosItemResponse<CosmosItemProperties> createResponse = container.createItem(cosmosItemProperties);
+        CosmosItemResponse<CosmosItemProperties> createResponse = this.container.createItem(cosmosItemProperties);
         String diagnostics = createResponse.getCosmosResponseDiagnostics().toString();
         assertThat(diagnostics).contains("\"connectionMode\":\"GATEWAY\"");
         assertThat(diagnostics).doesNotContain(("\"gatewayStatistics\":null"));
         assertThat(diagnostics).contains("\"operationType\":\"Create\"");
+        assertThat(diagnostics).contains("\"metaDataName\":\"ContainerLookUp\"");
+        assertThat(diagnostics).contains("\"serializationType\":\"PartitionKeyFetchSerialization\"");
         assertThat(createResponse.getCosmosResponseDiagnostics().getRequestLatency()).isNotNull();
         validateTransportRequestTimelineGateway(diagnostics);
     }
 
-    @Test(groups = {"simple"})
+    @Test(groups = {"simple"}, priority = 1)
```

Comment:
why do we need explicit `priority = 1` here?

Reply:
Because we want gatewayDiagnostics() and directDiagnostics() to run before gatewayDiagnosticsOnException and directDiagnosticsOnException(), other wise we wont able to test CONTAINER_LOOK_UP and PARTITION_KEY_RANGE_LOOK_UP with 100% of surety in either of these.